packages/dashboard/src/components/tasks/utils.ts

```diff
@@ -1,67 +1,87 @@
 import { SubmitTask } from 'api-client';
 import * as RmfModels from 'rmf-models';
 
+const genericErrors = {
+  checkArray: 'Expected an array of tasks',
+  checkTaskObject: 'Expected task to be an object',
+  defaultError: 'Unknown task type',
+};
+
+// errors associated with a certain type of task
+const customTaskFileErrors: Record<string, Record<string, string>> = {
+  // task fields expected of all task types
+  commonTaskFields: {
+    task_type: 'Expected task_type field to be a number',
+    start_time: 'Expected start_time field to be a number',
+    priority: 'Expected priority field to be a number',
+    description: 'Expected description field to be an object',
+  },
+  // error for clean task type
+  cleanTask: {
+    cleaning_zone: 'Cleaning Task - Expected cleaning_zone field to be a string',
+  },
+  // error for delivery task type
+  deliveryTask: {
+    pickup_place_name: 'Delivery Task - expected pickup_place_name to be a string',
+    pickup_dispenser: 'Delivery Task - expected pickup_dispenser to be a string',
+    dropoff_ingestor: 'Delivery Task - expected dropoff_ingestor to be a string',
+    dropoff_place_name: 'Delivery Task - expected dropoff_place_name to be a string',
+  },
+  // error for loop task type
+  loopTask: {
+    num_loops: 'Loop Task - expected num_loops to be a number',
+    start_name: 'Loop Task - expected start_name to be a string',
+    finish_name: 'Loop Task - expected finish_name to be a string',
+  },
+};
+
 /* istanbul ignore next */
 function checkField(
   obj: Record<string, unknown>,
   field: string,
   type: string,
-  errMsgs: string[],
+  taskType: 'commonTaskFields' | 'cleanTask' | 'deliveryTask' | 'loopTask',
 ): void {
-  try {
-    if (!Object.prototype.hasOwnProperty.call(obj, field) || typeof obj[field] !== type) {
-      throw new TypeError(`expected [${field}] to be [${type}]`);
-    }
-  } catch (err) {
-    errMsgs.push(err.message);
+  if (!Object.prototype.hasOwnProperty.call(obj, field) || typeof obj[field] !== type) {
+    throw new TypeError(customTaskFileErrors[taskType][field]);
   }
 }
 
-export interface SubmitTaskDetails {
-  submitTask: SubmitTask[];
-  errors: { [index: string]: string[] };
-}
-
 // TODO: See if we can generate validators from the schema.
 /* istanbul ignore next */
-export function parseTasksFile(contents: string): SubmitTaskDetails {
+export function parseTasksFile(contents: string): SubmitTask[] {
   const tasks = JSON.parse(contents);
-  let errMsgs: string[] = [];
-  let res: SubmitTaskDetails = { submitTask: [], errors: {} };
   if (!Array.isArray(tasks)) {
-    throw new TypeError('expected an array');
+    throw new TypeError(genericErrors.checkArray);
   }
 
   tasks.forEach((t, i) => {
```

Comment:
The way this works doesn't produce helpful error messages, it stops on the first error, and the error message does not easily identify which task contains the error.

I suggest something like this.
```ts
let errMsg = '';
tasks.forEach((task, idx) => {
  try {
    ...checkField();
  } catch (e) {
    errMsg += `Task ${idx + 1}: ${e.message}`;
  }
  try {
    ...checkField();
  } catch (e) {
    errMsg += `Task ${idx + 1}: ${e.message}`;
  }
  ...repeat
});
if (errMsg.length > 0) {
  throw new Error(errMsg);
}
```

To reduce the number of try/catch, `checkField` functions can be refactored to not throw and instead return a non-empty string if there is an error.

`customTaskFileErrors` is also not needed as the `checkField` function already contains all the information needed to return the error message. If the goal is to standardize error messages, you should define a custom error class instead.

```ts
class MyError extends Error {
  constructor(public fieldName: string, public expectedType: string) {
    super();
    this.message = 'my error message';
  }
}
```